src/lib/ActionClient.js

```diff
@@ -138,6 +138,15 @@ class ActionClient extends EventEmitter {
     });
     return id;
   }
+
+  shutdown() {
+    this.removeAllListeners();
+    this._goalPub.shutdown();
```

Comment:
could you return all these shutdowns in a `Promise.all()` ? Publishers and subscribers return promises on shutdown so that callers can know when they are fully destroyed (unregistered from the ROS master, clients disconnected, etc)